### PR TITLE
Expose tracksViewChange and icon property to improve some performance issues

### PR DESCRIPTION
**Pull request by PhamBaTho**
Expose [tracksViewChanges](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_marker.html#abaca10e1f2ba33aa1a10d7014269e780) property of GMSMarker. By default, this property is YES. 

Setting `tracksViewChange` to NO appropriately can solve **some performance issues** causing high CPU load on iOS (e.g. #1031 , #1176).

The followings are some guidelines relating to `tracksViewChange` I extracted from [Google Maps SDK documentation](https://developers.google.com/maps/documentation/ios-sdk/marker):

> - The UIView can be demanding on resources when tracksViewChanges is set to YES, which may result in **increased battery usage**. In comparison, a single frame UIImage is static and does not need to be re-rendered.
> - **Some devices may struggle to render the map if you have many markers on screen**, and each marker has its own UIView, and all markers are tracking changes at the same time.
> - To decide when to set the tracksViewChanges property, you **should weigh up performance** considerations against the advantages of having the marker redrawn automatically. For example:
>   * If you have a series of changes to make, you can change the property to YES then back to NO.
>   * When an animation is running or the contents are being loaded asynchronously, you should keep the property set to YES until the actions are complete.
> 

Please note that this property has no effect if iconView of marker is nil. In other words, I don't think we have any issue with performance if we use default marker.

I have tested on iOS simulator with many use cases like adding many markers on maps, drawing polyline between markers, animating custom markers... and the CPU remains in good health when the map is idle.

Sample usage:
```
<MapView.Marker
   tracksViewChanges={ false }
   anchor={ { x: 1, y: 1 } }
   coordinate={ {
      longitude: location.longitude,
      latitude: location.latitude,
   } }>
   { this.renderMarker() }
</MapView.Marker>
```

### CODE REVIEW

**Comment by PhamBaTho** [comment]
UPDATE:
I have just updated this PR a bit. I'm exposing additionally [icon](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_marker.html#af709ab9605990b2a1c616be8c88eddbc) property. It's equivalent to `icon` property of GMSMarker Class. 

The `image` property is great, it can cover most of the use case. Though, I believe that using `icon` property still has an advantage over `image` property in term of performance, battery usage... because it doesn't trigger tracksViewChanges which [has no effect if iconView is nil](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_marker.html#abaca10e1f2ba33aa1a10d7014269e780).

We still need to have a better naming between `icon` and existing `image` to make them less confusing.

File: lib/components/MapMarker.js [hunk 1]
```diff
@@ -260,13 +273,20 @@ class MapMarker extends React.Component {
       image = image.uri;
     }
 
+    let icon;
+    if (this.props.icon) {
```
Comment (henrikra) [inline lib/components/MapMarker.js]: I would refactor this to ternary if statement :)

**Comment by heysailor** [comment]
Thanks for this. Could you give a full example of use with the icon property?

**Comment by juergengunz** [comment]
i discovered that if i use <image> the images are not being displayed on google maps (ios)... when i use the image property it works fine but i am not able to resize the images

**Comment by dragonfire1119** [comment]
Seems like the icon prop doesn't do anything? If I try to use a regular image in icon prop it just shows the regular red icon from google maps. Is it suppose to work with a Custom Image like a .png? Also I set tracksViewChanges={ false } & it didn't change anything. The CPU is still at 90%+

**Comment by krabbypattified** [comment]
+1 would really like to see these optimizations in my app!

**Comment by compojoom** [comment]
@PhamBaTho - how do I add your changes to an existing project? I tried directly modifying the AirGoogleMaps in the project in xCode, but then I get a compile error:
"Linker command failed with exit code 1"

File: docs/marker.md [hunk 2]
```diff
@@ -17,6 +18,7 @@
 | `identifier` | `String` |  | An identifier used to reference this marker at a later date.
 | `rotation` | `Float` |  | A float number indicating marker's rotation angle, in degrees.
 | `draggable` | `<null>` |  | This is a non-value based prop. Adding this allows the marker to be draggable (re-positioned).
+| `tracksViewChanges` | `Boolean` |  | Controls whether the icon for this marker should be redrawn every frame.
```
Comment (gpeal) [inline docs/marker.md]: @PhamBaTho Is this prop iOS only?

File: lib/components/MapMarker.js [hunk 3]
```diff
@@ -50,6 +50,14 @@ const propTypes = {
   image: PropTypes.any,
 
   /**
+   * Marker icon to render (equivalent to `icon` property of GMSMarker Class).
+   * Using this property has an advantage over `image` in term of performance, battery usage...
+   * because it doesn't trigger tracksViewChanges.
+   * (tracksViewChanges is set to YES by default if iconView is not nil).
+   */
+  icon: PropTypes.any,
```
Comment (gpeal) [inline lib/components/MapMarker.js]: @PhamBaTho Can you give this a more specific prop type?

File: lib/components/MapMarker.js (hunk 1 above)
Comment (gpeal) [reply lib/components/MapMarker.js]: +1

File: docs/marker.md (hunk 2 above)
Comment (satori-ytolstoguzov) [reply docs/marker.md]: Only applicable for iOS. Android don't have issues with that.

**Comment by satori-ytolstoguzov** [comment]
At what stage this PR? I use a fork with `tracksViewChanges` in my project for at least half a year while waiting to be merged.
@PhamBaTho Are you going to fix comments?

**Comment by rborn** [comment]
@satori-ytolstoguzov if you could address the comments (if @PhamBaTho won't) I can have a look

**Comment by compojoom** [comment]
@rborn @satori-ytolstoguzov @PhamBaTho - we've merged this back in November: https://github.com/react-community/react-native-maps/pull/1705 and it exposes tracksViewChanges. Shouldn't we close this pull request? Or what is the difference here with pull request 1705?

**Comment by satori-ytolstoguzov** [comment]
@compojoom It seems merged PR forgot to add that to the docs. Other than that I don't see any differences

**Comment by rborn** [comment]
@satori-ytolstoguzov could you test if it's working and make a PR for documentation? I'll be here to merge it ❤️

**Comment by satori-ytolstoguzov** [comment]
@rborn It's a busy day, will look into that on Monday. Also would you consider useful if I add one more bridged property: `tracksInfoWindowChanges`. It's only for iOS and without it Google Maps SDK ignores all changes in callout after first render.

**Comment by pqminh** [comment]
i'm testing on maps provider google and default.
rn 0.47.2
react-native-maps 0.17.1 | 0.19.0. 
All of them don't working. I'm render maps about 700 markers. When use provider google, it's so slow and lag. And use provider default. It does not slow, but when press on random marker, callout is show behind the markers around...

**Comment by christopherdro** [comment]
trackViewChange prop has been added here: 78a36ac3727d9781187b008d255fc365b7873f94

**Pull request closed without merging**